src/unitxt/task.py

```diff
@@ -10,42 +13,141 @@ class Tasker:
+        if isinstance(data_type, str):
+            data_type = data_type.replace("typing.", "")
+            return parse_type_string(data_type)
+        return data_type
+
     def verify(self):
+        for io_type in ["inputs", "outputs"]:
+            data = self.inputs if io_type == "inputs" else self.outputs
+            if not isinstance(data, Dict):
+                get_logger().warning(
+                    f"'{io_type}' should be a dict of field names and their types, instead only "
+                    f"'{data}' was passed. All types will be assumed to be 'Any'. In future version "
+                    f"of unitxt this will raise an exception."
+                )
+                data = {key: Any for key in data}
+                if io_type == "inputs":
+                    self.inputs = data
+                else:
+                    self.outputs = data
+
+        if not self.prediction_type:
+            get_logger().warning(
+                "'prediction_type' was not passed. It must be compatible with used metrics. "
+                "Trying to infer `prediction_type` based on specified metrics. In future "
+                "version of unitxt this will raise an exception."
+            )
+            sample_metric = fetch_artifact(self.metrics[0])[0]
+            self.prediction_type = sample_metric.get_prediction_type()
+
         for augmentable_input in self.augmentable_inputs:
             assert (
                 augmentable_input in self.inputs
             ), f"augmentable_input f{augmentable_input} is not part of {self.inputs}"
 
-    def process(
-        self, instance: Dict[str, Any], stream_name: Optional[str] = None
-    ) -> Dict[str, Any]:
-        try:
-            inputs = {key: instance[key] for key in self.inputs}
-        except KeyError as e:
-            raise KeyError(
-                f"Unexpected FormTask input column names ({[key for key in self.inputs if key not in instance]})."
-                f"The available input names: {list(instance.keys())}"
-            ) from e
+    def check_metrics_type(self) -> None:
+        for metric_name in self.metrics:
+            metric = fetch_artifact(metric_name)[0]
+            metric_prediction_type = metric.get_prediction_type()
+
+            if self.prediction_type == metric_prediction_type:
+                continue
+
+            prediction_type = self.process_type(self.prediction_type)
+            metric_prediction_type = self.process_type(metric_prediction_type)
+            if (
+                prediction_type == metric_prediction_type
+                or prediction_type == Any
+                or metric_prediction_type == Any
+            ):
+                continue
+
+            raise ValueError(
+                f"Given prediction type '{prediction_type}' and metric '{metric_name}' "
+                f"prediction type '{metric_prediction_type}' are different."
+            )
+
+    def access_instance_value(
```

Comment:
Maybe it's a style, but I think it would be clearer to have two separate functions

get_input_value(instance, key)
get_output_value(instance_key)

The erro message will be slightly repeated, but the code would be clearer.